File: android/src/main/kotlin/com/example/dart_xray/DartXrayPlugin.kt
package com.example.dart_xray

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Platform plugin registration for dart_xray.
 *
 * Engine lifecycle and status now use Dart FFI directly; no MethodChannel/EventChannel
 * bridge is registered for engine control.
 */
class DartXrayPlugin : FlutterPlugin {
  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    // Intentionally empty.
    // Android-side code in this package is reserved for OS concerns such as VpnService.
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    // Intentionally empty.
  }
}

File: ios/Classes/DartXrayPlugin.swift
import Flutter
import UIKit

/// iOS plugin registrar for dart_xray.
///
/// Engine control/status is FFI-backed in Dart and does not use platform channels.
public class DartXrayPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    _ = DartXrayPlugin()
  }
}
